src/components/Skeleton/Skeleton.tsx

```diff
@@ -0,0 +1,119 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './Skeleton.module.css';
+
+type BaseProps = {
+  /**
+   * CSS class names to augment the layout of the skeleton component. For skeletons,
+   * this is mostly useful to add layout styles (e.g., tailwind CSS margins, flex, etc.)
+   */
+  className?: string;
+  /**
+   * Determine whether there should be an animation in the skeleton state
+   */
+  isAnimating?: boolean;
+};
+
+type SkeletonProps = BaseProps & {
+  width?: number | string;
+  height?: number | string;
+};
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * `import {Skeleton} from "@chanzuckerberg/eds";`
+ *
+ * Skeleton states allow for hinting at the layout of pending content before the
+ * content is available. The different shapes allow for representation of different
+ * types of content. Mix and match to create a layout that mimicks the pending data.
+ *
+ * Skeletons come in a variety of shapes. The default is a round rect, and there are
+ * others for circular objects, or text layouts. Use any meaningful CSS length values
+ * (e.g., rem, ch, px, etc.)
+ *
+ * Examples:
+ *
+ * For text, Use `Skeleton.Text` and specify the lineHeight and the width (for example in 'ch' units).
+ *
+ * For circular objects, use `Skeleton.Circle' and specify the size.
+ *
+ * For all other shapes, use `Skeleton.Rect` (or just Skeleton) and specify the width and height.
+ */
+export const Skeleton = ({
+  className,
+  isAnimating = true,
+  height,
+  width,
+  ...other
+}: SkeletonProps) => {
+  const componentClassName = clsx(
+    className,
+    styles['skeleton'],
+    isAnimating && styles['skeleton--is-animating'],
+  );
+  return (
+    <div
+      aria-hidden
+      className={componentClassName}
+      {...other}
+      style={{ width, height }}
+    ></div>
+  );
+};
+
+type TextProps = BaseProps & {
+  lineHeight?: number | string;
+  width?: number | string;
+};
+
+const TextSkeleton = ({
+  className,
+  isAnimating = true,
+  lineHeight,
+  width,
+  ...other
+}: TextProps) => {
+  const componentClassName = clsx(
+    className,
+    styles['skeleton'],
+    isAnimating && styles['skeleton--is-animating'],
+  );
+  return (
+    <div
+      aria-hidden
+      className={componentClassName}
+      style={{ width, height: lineHeight }}
+    ></div>
+  );
+};
+
+type CirlceProps = BaseProps & {
+  size?: number | string;
```

Comment:
I wonder if `Circle` should use `width` instead of `size` to be consistent with the other shapes.

Reply:
I think this makes sense. I def intended to avoid people having circles that ... werent :)